services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/resource_tracker_credit_transactions.py

```diff
@@ -0,0 +1,87 @@
+from datetime import datetime
+from typing import Annotated
+
+from fastapi import Depends
+from models_library.api_schemas_resource_usage_tracker.credit_transactions import (
+    WalletTotalCredits,
+)
+from models_library.products import ProductName
+from models_library.resource_tracker import (
+    CreditTransactionId,
+    TransactionBillingStatus,
+    TransactionClassification,
+)
+from models_library.users import UserID
+from models_library.wallets import WalletID
+from pydantic import BaseModel
+
+from ..api.dependencies import get_repository
+from ..models.resource_tracker_credit_transactions import CreditTransactionCreate
+from ..modules.db.repositories.resource_tracker import ResourceTrackerRepository
+
+
+class CreditTransactionCreateBody(BaseModel):
+    product_name: ProductName
+    wallet_id: WalletID
+    wallet_name: str
+    user_id: UserID
+    user_email: str
+    credits: float
+    payment_transaction_id: str
+    created_at: datetime
+
+
+async def create_credit_transaction(
+    credit_transaction_create_body: CreditTransactionCreateBody,
+    resource_tracker_repo: Annotated[
+        ResourceTrackerRepository, Depends(get_repository(ResourceTrackerRepository))
+    ],
+) -> CreditTransactionId:
+    transaction_create = CreditTransactionCreate(
+        product_name=credit_transaction_create_body.product_name,
+        wallet_id=credit_transaction_create_body.wallet_id,
+        wallet_name=credit_transaction_create_body.wallet_name,
+        pricing_plan_id=None,
+        pricing_detail_id=None,
+        user_id=credit_transaction_create_body.user_id,
+        user_email=credit_transaction_create_body.user_email,
+        credits=credit_transaction_create_body.credits,
+        transaction_status=TransactionBillingStatus.BILLED,
+        transaction_classification=TransactionClassification.ADD_WALLET_TOP_UP,
+        service_run_id=None,
+        payment_transaction_id=credit_transaction_create_body.payment_transaction_id,
+        created_at=credit_transaction_create_body.created_at,
+        last_heartbeat_at=credit_transaction_create_body.created_at,
+    )
+    transaction_id = await resource_tracker_repo.create_credit_transaction(
+        transaction_create
+    )
+
+    # Shoot and forget
+    wallet_total_credits = (
+        await resource_tracker_repo.sum_credit_transactions_by_product_and_wallet(
+            credit_transaction_create_body.product_name,
+            credit_transaction_create_body.wallet_id,
+        )
+    )
+    assert wallet_total_credits  # nosec
+    # TODO: Publish wallet total credits to RabbitMQ
```

Comment:
I understand generally that there was a decision not to use TODOs because they were left in the code for a long time. In my case, this is something that needs to be implemented straight away in the next PR (therefore TODO makes sense in this use-case for me, but I can change it)